File: final_projects/ROS1_to_ROS2_Migration_Project/project_ros1_noetic/src/turtlesim_project_cpp/src/target_publisher.cpp
#include <ros/ros.h>
#include <my_robot_msgs/Coordinates2D.h>
#include <cstdlib>

class TargetPublisher
{

public:
    TargetPublisher(ros::NodeHandle *nh)
    {
        double publish_interval;
        ros::param::param<double>("~publish_interval", publish_interval, 3.0);
        pub_ = nh->advertise<my_robot_msgs::Coordinates2D>("target_coordinates", 10);
        timer_ = nh->createTimer(ros::Duration(publish_interval), std::bind(&TargetPublisher::sendRandomCoordinates, this));
        ROS_INFO("Target publisher has been started");
    }

private:
    ros::Publisher pub_;
    ros::Timer timer_;

    // returns random double number in range [0.0, 1.0)
    double randomDouble()
    {
        return double(std::rand()) / (double(RAND_MAX) + 1.0);
    }

    void sendRandomCoordinates()
    {
        double x = randomDouble() * 11.0;
        double y = randomDouble() * 11.0;
        publishCoordinates(x, y);
    }

    void publishCoordinates(double x, double y)
    {
        my_robot_msgs::Coordinates2D msg;
        msg.x = x;
        msg.y = y;
        pub_.publish(msg);
    }
};

int main(int argc, char **argv)
{
    ros::init(argc, argv, "target_publisher");
    ros::NodeHandle nh;
    TargetPublisher target_publisher = TargetPublisher(&nh);
    ros::spin();
    return 0;
}

File: INTERNSHIP/final_projects/ROS1_to_ROS2_Migration_Project/project_ros1_noetic/src/turtlesim_project_py/scripts/target_publisher.py
#!/usr/bin/env python3
import rospy
import random
from my_robot_msgs.msg import Coordinates2D


class TargetPublisher:

    def __init__(self):
        publish_interval = rospy.get_param("~publish_interval", 3.0)
        self.pub_ = rospy.Publisher(
            "target_coordinates", Coordinates2D, queue_size=10)
        self.timer_ = rospy.Timer(rospy.Duration(
            publish_interval), self.send_random_coordinates)
        rospy.loginfo("Target publisher has been started")

    def publish_coordinates(self, x, y):
        msg = Coordinates2D()
        msg.x = x
        msg.y = y
        self.pub_.publish(msg)

    def send_random_coordinates(self, event=None):
        x = random.uniform(0.0, 11.0)
        y = random.uniform(0.0, 11.0)
        self.publish_coordinates(x, y)


if __name__ == '__main__':
    rospy.init_node('target_publisher')
    coordinates_generator = TargetPublisher()
    rospy.spin()
